src/together/utils/tools.py

```diff
@@ -23,18 +25,62 @@ def normalize_key(key: str) -> str:
     return key.replace("/", "--").replace("_", "-").replace(" ", "-").lower()
 
 
-def parse_timestamp(timestamp: str) -> datetime:
-    formats = ["%Y-%m-%dT%H:%M:%S.%fZ", "%Y-%m-%dT%H:%M:%SZ"]
-    for fmt in formats:
-        try:
-            return datetime.strptime(timestamp, fmt)
-        except ValueError:
-            continue
-    raise ValueError("Timestamp does not match any expected format")
+def parse_timestamp(timestamp: str) -> datetime | None:
+    """Parse a timestamp string into a datetime object or None if invalid.
+
+    Args:
+        timestamp (str): Timestamp in ISO 8601 format (e.g. "2021-01-01T00:00:00Z")
+
+    Returns:
+        datetime | None: Parsed datetime, or None if the string is empty
+    """
+    if timestamp == "":
+        return None
+
+    return datetime.fromisoformat(timestamp.replace("Z", "+00:00"))
+
+
+def format_event_timestamp(event: Any) -> str:
+    """Format event timestamp to a readable date string.
+
+    Args:
+        event: An event object with a created_at attribute
+
+    Returns:
+        str: Formatted timestamp string (MM/DD/YYYY, HH:MM AM/PM)
+    """
+    timestamp = parse_timestamp(event.created_at or "")
+    return timestamp.strftime("%m/%d/%Y, %I:%M %p") if timestamp else ""
+
+
+def get_event_step(event: Any) -> str | None:
```

Comment:
What do you think about adding a few unit tests to ensure all expected scenarios are covered correctly?

Reply:
Rewrote the function -- I don't think it needs unit tests for now on